### PR TITLE
Set up CI with Azure Pipelines for deploying permission scraper

### PR DESCRIPTION
**Pull request by MIchaelMainer**

### CODE REVIEW

**Review by irvinesunday — DISMISSED** [review]

File: .azure-pipelines/deploy-permissions-scraper.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# .NET Core Function App to Windows on Azure
+# Build a .NET Core function app and deploy it to Azure as a Windows function App.
+# Add steps that analyze code, save build artifacts, deploy, and more:
+# https://docs.microsoft.com/en-us/azure/devops/pipelines/languages/dotnet-core
+
+trigger:
+- master
+
```
Comment (MIchaelMainer) [inline .azure-pipelines/deploy-permissions-scraper.yml:16]: What path spec should we define?

File: .azure-pipelines/deploy-permissions-scraper.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    pool:
+      vmImage: $(vmImageName)
+
+    steps:
+    - task: DotNetCoreCLI@2
+      displayName: Build
+      inputs:
+        command: 'build'
+        projects: |
+          $(workingDirectory)/*.csproj
```
Comment (MIchaelMainer) [inline .azure-pipelines/deploy-permissions-scraper.yml]: We should define the specific projects we care about for deployment.

File: .azure-pipelines/deploy-permissions-scraper.yml [hunk 3]
```diff
@@ -0,0 +1,76 @@
+# .NET Core Function App to Windows on Azure
+# Build a .NET Core function app and deploy it to Azure as a Windows function App.
+# Add steps that analyze code, save build artifacts, deploy, and more:
+# https://docs.microsoft.com/en-us/azure/devops/pipelines/languages/dotnet-core
+
```
Comment (MIchaelMainer) [inline .azure-pipelines/deploy-permissions-scraper.yml:5]: Let's add a verbose naming convention that helps when having to run this multiple times.

`name: $(BuildDefinitionName)_$(SourceBranchName)_$(Date:yyyyMMdd)$(Rev:.r)`

File: .azure-pipelines/deploy-permissions-scraper.yml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      inputs:
+        rootFolderOrFile: '$(System.DefaultWorkingDirectory)/publish_output'
+        includeRootFolder: false
+        archiveType: zip
+        archiveFile: $(Build.ArtifactStagingDirectory)/$(Build.BuildId).zip
+        replaceExistingArchive: true
+
+    - publish: $(Build.ArtifactStagingDirectory)/$(Build.BuildId).zip
+      artifact: drop
+
```
Comment (MIchaelMainer) [inline .azure-pipelines/deploy-permissions-scraper.yml:62]: We should add a notification for when this pipelines fails when run automatically. We may need to setup this task in this ADO project.

File: .azure-pipelines/deploy-permissions-scraper.yml (hunk 1 above)
Comment (MIchaelMainer) [reply .azure-pipelines/deploy-permissions-scraper.yml:16]:
```yaml
trigger:
  branches:
    include:
    - master
  paths:
    exclude:
    - .gitignore
    - LICENSE
```

File: .azure-pipelines/deploy-permissions-scraper.yml (hunk 4 above)
Comment (MIchaelMainer) [reply .azure-pipelines/deploy-permissions-scraper.yml:62]:
```yaml
  - task: YodLabs.O365PostMessage.O365PostMessageBuild.O365PostMessageBuild@0
    displayName: 'Graph Client Tooling pipeline fail notification'
    inputs:
      addressType: serviceEndpoint
      serviceEndpointName: 'microsoftgraph pipeline status'
      title: '$(Build.DefinitionName) failure notification'
      text: 'This pipeline has failed. View the build details for further information. This is a blocking failure. '
    condition: and(failed(), ne(variables['Build.Reason'], 'Manual'))
    enabled: true
```

Comment (MIchaelMainer) [reply .azure-pipelines/deploy-permissions-scraper.yml:62]: Create an issue to track adding this.

Comment (irvinesunday) [reply .azure-pipelines/deploy-permissions-scraper.yml:62]: Do we add this once we've already setup the task in ADO?

Comment (MIchaelMainer) [reply .azure-pipelines/deploy-permissions-scraper.yml:62]: I've requested this task from DeVere. Let me see if we can get this added today.

**Review by irvinesunday — DISMISSED** [review]

**Review by irvinesunday — APPROVED** [review]

**Pull request merged**